### ISSUE TITLE
Jenkins Testing: Dynamic parallel should support numMachines per test target

### ISSUE THREAD

**Issue by AdamBrousseau** [adoptium/ci-jenkins-pipelines#184]
https://github.com/adoptium/ci-jenkins-pipelines/pull/142 introduced parallel testing to the pipelines.
```
 "defaultDynamicParas": {
            "testLists"      : ["extended.openjdk"],
            "numMachines"    : "3"
```
At the moment users can supply an array of test targets and a single numMachines value. We have use case to have different numMachines for different targets.

**Comment by AdamBrousseau** [comment]
An override via the spec file to change the number for a given platform might be valuable too.
Eg. extended.openjdk => 3 machines except for Windows set to 5 for example.

**Comment by AdamBrousseau** [comment]
Resolved by #213
